main.go

```diff
@@ -1,227 +1,200 @@
-				Hash:                  "88560a7f9a4e54f7a8d83313dad6be31cb986d2f",
-			},
-		},
-		PR: PullRequest{
-			URL: "https://github.com/src-d/lookout-test-fixtures/pull/13",
+		URL: PullRequestURL{
+			Owner:  "src-d",
+			Repo:   "lookout-test-fixtures",
+			Number: 13,
 		},
+		CurrentRevision: 0,
 	},
 }
 
-// PullRequest is a struct with information about pull request
-type PullRequest struct {
-	URL string
-	// add recorded responses here later?
+// PullRequestURL is a struct with information about a pull request URL:
+// github.com/<Owner>/<Repo>/pull/<Number>
+type PullRequestURL struct {
+	Owner  string
+	Repo   string
+	Number int
 }
 
 // Fixture is struct for a test case
 type Fixture struct {
-	Name           string
-	CommitRevision pb.CommitRevision
-	PR             PullRequest
+	Name string
+	URL  PullRequestURL
+	// There can be more than one revision of the fixture, but it should have
+	// files for 0..CurrentRevision
+	CurrentRevision int
+}
+
+// GetRevision returns a copy of this Fixture at a different revision
+func (f Fixture) GetRevision(revision int) *Fixture {
+	copy := f
+	copy.CurrentRevision = revision
+
+	return &copy
+}
+
+// GetRevisions returns all this Fixture revisions, ordered from 0 to CurrentRevision
+func (f Fixture) GetRevisions() []*Fixture {
+	list := make([]*Fixture, f.CurrentRevision+1)
+	for v := 0; v <= f.CurrentRevision; v++ {
+		list[v] = f.GetRevision(v)
+	}
+
+	return list
+}
+
+// GetPR returns the fixture saved from the github.Client.PullRequests.Get
+// output
+func (f Fixture) GetPR() (*github.PullRequest, error) {
+	filename := f.FilenamePR(f.CurrentRevision)
+	data, err := Asset(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	var pr github.PullRequest
+	err = json.Unmarshal(data, &pr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pr, nil
+}
+
+// GetCommitsComparison returns the fixture saved from the
+// github.Client.Repositories.CompareCommits output
+func (f Fixture) GetCommitsComparison() (*github.CommitsComparison, error) {
+	filename := f.FilenameCC(f.CurrentRevision)
+	data, err := Asset(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	var cc github.CommitsComparison
+	err = json.Unmarshal(data, &cc)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cc, nil
 }
 
-// Fixtures is a list of fixtures
+func (f Fixture) FilenamePR(revision int) string {
+	return filepath.FromSlash(fmt.Sprintf("fixtures/%s.%s.pr-%d-v%d.json", f.URL.Owner, f.URL.Repo, f.URL.Number, revision))
+}
+
+func (f Fixture) FilenameCC(revision int) string {
+	return filepath.FromSlash(fmt.Sprintf("fixtures/%s.%s.pr-%d-v%d-compare.json", f.URL.Owner, f.URL.Repo, f.URL.Number, revision))
+}
+
+// Fixtures is a list of fixtures. If a fixture has more than one revision the
+// most recent one is used
 type Fixtures []*Fixture
 
-// GetByName returns fixture by name
+// GetByName returns fixture by name. If the fixture has more than one revision
+// the most recent one is used
```

Comment:
> the most recent one is used

I don't see any code for that. Does it mean we have to maintain the list in correct order?
Then the comment should be `If the fixture has more than one revision the one which appears first in the list will be used`

Reply:
ah. Got it. We need to update CurrentRevision when we add one more revision.

Reply:
exactly